web/scripts/app.js

```diff
@@ -4,6 +4,7 @@
 var grid; // The SlickGrid.
 var sortCol = {id: 'wilks'}; // Initial column sorting information.
 var sortAsc = false; // Initial column sorting information.
+var searchInfo = {lastrow: 0, laststr: ''};
```

Comment:
Hm, I actually think this behavior with `lastrow` might be unexpected. Searches in office tools have the concept of "the current point", like a text selection, from which searching proceeds. So if you execute a search, then click some text behind the result and execute the search again, you would expect to get the same result.

This implementation ignores the document state and always proceeds linearly. So if for example someone searches for a name, then scrolls up a little bit and re-executes the search, they would get something else!

Probably this can be fixed by removing the concept of `lastrow` and instead executing the search from the top of the grid plus one row. You can ask the grid for the number of the top row via `grid.getViewport()`, which will return an object with the property `top` set to the top visible row`s ID.